### PR TITLE
Fix print_usage and man page mistakes

### PR DESCRIPTION
**Pull request by 4G3NT**
Usage string is off and this PR fixes it.

### CODE REVIEW

**Review by kbdharun — APPROVED** [review]

**Comment by SethFalco** [comment]
Just for context, I checked what the client specifications had on this just to make sure we're changing the correct thing.

Indeed, `-v` should be for `--version`.

**Review by blueskyson — APPROVED** [review]
Look good to me. Maybe @MasterOdin or @Leandros can take a look.

**Comment by 4G3NT** [comment]
Turns out that other options, such as `--verbose` and `--version`, do exist, on top of `-l` being non-existent.

img

**Comment by 4G3NT** [comment]
Hopefully this will be the last commit. :slightly_smiling_face:

**Comment by 4G3NT** [comment]
If I have time, I can rewrite the arguments so that `--version` is the only flag, and `-v`, `--verbose` is used for verbose.

**Comment by 4G3NT** [comment]
Not sure what to do with the current usage string. Maybe it should be:

**tldr** [_OPTION_] _PAGE_

or **tldr** [ **--verbose**] [**-c**|**-u**] [_OPTION_]... _PAGE_

But the problem is it doesn't cover options like 'render,' which takes a file path, for example.

**Comment by 4G3NT** [comment]
Not sure what to do with the current usage string. Maybe it should be:

**tldr** [_OPTION_] _PAGE_

or **tldr** [ **--verbose**] [**-c**|**-u**] [_OPTION_]... _PAGE_

But the problem is it doesn't cover options like `--render`, which takes a file path, for example.

**Comment by 4G3NT** [comment]
Reverting to the old usage but using `tldr` [_--verbose_] [_OPTION_]... _PAGE_ is much better IMO.

**Review by kbdharun — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

File: src/tldr.c [hunk 1]
```diff
@@ -149,8 +154,8 @@ int main(int argc, char **argv) {
         return EXIT_SUCCESS;
     }
     if (verbose_flag && optind >= argc) {
-        print_version(argv[0]);
-        return EXIT_SUCCESS;
+        print_usage(argv[0]);
+        return EXIT_FAILURE;
```
Comment (sbrl) [inline src/tldr.c:158]: I assume this exits with code e.g. `1` instead of `0` when printing help? Other commands e.g. `ls`, `tar`, etc exit with code `0` when printing help, so we should do the same.

Unless I misunderstood this?

File: src/tldr.c [hunk 2]
```diff
@@ -82,12 +82,13 @@ int main(int argc, char **argv) {
...
 
         case 'v':
-            verbose_flag = 1;
+            version_flag = 1;
             break;
 
         case '?':
             /* do not set the help flag, only show getopt error */
             /* help_flag = 1; */
+            return EXIT_FAILURE;
```
Comment (sbrl) [inline src/tldr.c:91]: See below comment about exiting with code `0` instead of `1` being preferable when printing help.

**Review by sbrl — COMMENTED** [review]
Just a quick comment or two about behaviour when writing the help. If you do e.g. `ls --help`, it exits with code 0. This seems to be a well established pattern, so I'd recommend we do the same here in the Node client.

File: src/tldr.c (hunk 1 above)
Comment (kbdharun) [reply src/tldr.c:158]: Our client specification v1.4 suggests to use a [non-zero error code](https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md) for finding pages. (https://github.com/tldr-pages/tldr/pull/4246)

Not sure if we should go with it here.

Comment (sbrl) [reply src/tldr.c:158]: \* Specifically, a client MUST exit with a non-zero exit code if the page could not be found anywhere.

Comment (4G3NT) [reply src/tldr.c:158]: > I assume this exits with code e.g. `1` instead of `0` when printing help? Other commands e.g. `ls`, `tar`, etc exit with code `0` when printing help, so we should do the same.
\
Unless I misunderstood this?

This basically exits with `1` if invoked with `tldr` instead of `tldr --help`, which would exit with `0`.

In other words, if invoking `tldr` to display help on its own, rather than the user invoking it, the program would exit with `1`.

Example:
```sh
$ tldr > /dev/null; echo $?
1
$ tldr --help > /dev/null; echo $?
0
```

Comment (4G3NT) [reply src/tldr.c:158]: > I assume this exits with code e.g. `1` instead of `0` when printing help? Other commands e.g. `ls`, `tar`, etc exit with code `0` when printing help, so we should do the same.
\
> Unless I misunderstood this?

This exits with `1` on a getopt error.

For example:
* `tldr -a`, which is an unknown argument, exits with `1` and displays a getopt error.
* `tldr -v` exits with `0` and displays the version.

Comment (4G3NT) [reply src/tldr.c:158]: You can even test this by running these commands:

```sh
$ tldr -a 2> /dev/null; echo $?
1
$ tldr -v > /dev/null; echo $?
0
```

File: src/tldr.c [hunk 3]
```diff
@@ -210,25 +215,25 @@ void print_version(char const *arg) {
...
+    fprintf(stdout, "    %-23s %s\n", "-h, --help", "print this help and exit");
+    fprintf(stdout, "    %-23s %s\n", "-p, --platform=PLATFORM",
             "select platform, supported are linux / osx / sunos / windows / common");
-    fprintf(stdout, "    %-20s %-30s\n", "--linux", "show command page for Linux");
-    fprintf(stdout, "    %-20s %-30s\n", "--osx", "show command page for OSX");
-    fprintf(stdout, "    %-20s %-30s\n", "--sunos", "show command page for SunOS");
-    fprintf(stdout, "    %-20s %-30s\n", "-r, --render=PATH",
+    fprintf(stdout, "    %-23s %s\n", "-r, --render=PATH",
             "render a local page for testing purposes");
-    fprintf(stdout, "    %-20s %-30s\n", "-C, --color", "force color display");
```
Comment (4G3NT) [inline src/tldr.c:232]: `%-23s` should be used and `%-30s` isn't needed as far as I know.

File: src/tldr.c (hunk 1 above)
Comment (sbrl) [reply src/tldr.c:158]: Ah okay. So can you confirm that when you call e.g. `tldr --help` it exits with code `0`, @4G3NT?

Comment (4G3NT) [reply src/tldr.c:158]: > Ah okay. So can you confirm that when you call e.g. `tldr --help` it exits with code `0`, @4G3NT?

Yes, it exits with code `0` because of this code:

```c
if (help_flag) {
    print_usage(argv[0]);
    return EXIT_SUCCESS;
}
```

**Comment by 4G3NT** [comment]
And now when you invoke `tldr` with `-r` it will output:

```sh
$ tldr -r
./tldr: option requires an argument -- 'r'
```
instead of the last code:
```sh
$ tldr -r
tldr: option '-render' requires an argument
```

**Review by kbdharun — APPROVED** [review]

File: src/tldr.c (hunk 1 above)
Comment (sbrl) [reply src/tldr.c:158]: Very cool, many thanks!

**Review by sbrl — APPROVED** [review]

**Comment by sbrl** [comment]
Not sure who is generally responsible for the c client but this seems ready to merge for me.
